source/PlanetPanel.cpp

```diff
@@ -297,24 +298,20 @@ void PlanetPanel::TakeOffIfReady()
 				out << " of your mission cargo.";
 			}
 		}
+		else if (fighterCount > 0 || droneCount > 0)
+		{
+			int nonJumpCount = fighterCount > 0 ? fighterCount : 0;
+			nonJumpCount += droneCount > 0 ? droneCount : 0;
+			out << "If you take off now you will launch with ";
+			if (nonJumpCount == 1)
+				out << "a ship";
+			else
+				out << nonJumpCount << " ships";
+			out << " that will not be able to leave the system.";
```

Comment:
Why not just show this warning for all ships that can't jump instead of just counting uncarried fighters and drones?

Reply:
You're right. The reason I didn't do that yet was to keep this pull request simple.
I'll push a commit that also checks JumpsRemaining() of each ship in the loop.